### PR TITLE
[Feature] Implement evil teams

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes?
Enables evil team grunts and leader to spawn

## Why am I doing these changes?
* Evil teams are a commonly requested feature
* Classic mode is lacking content in the midgame apart from a few rival battles
* Players only need to prepare for the final boss and the rival as the main hurdles in a classic playthrough. This introduces some roguelike randomness where there is another major boss that players have to prepare for

## What did change?
* Uncommented lines in classicFixedBattles
* Changed the `getEvilGruntPartyTemplate` function so it would no longer crash since it checks for the waveIndex of both 61 and 62 for an encounter on floor 62. Switched to just using if/else instead.

## How to test the changes?
Start a classic run and check out the new encounters on floors 35, 62, 64, 66 ,112, 114, 115, and 165
You can also use overrides to jump to those floors

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by HopsWas** [comment]
tested a few runs, appears to be working properly.

**Comment by returntoice** [comment]
I'm really looking forward to this update.

**Comment by CodeTappert** [comment]
The grunts definitly need more dialouge options

**Comment by CodeTappert** [comment]
> The grunts definitly need more dialouge options

Otherwise it works as described. No comment from me if this is balanced etc. Because i dont know. It wasnt too hard for me

File: src/locales/pt_BR/dialogue.ts [hunk 1]
```diff
@@ -383,6 +383,186 @@ export const PGMdialogue: DialogueTranslationEntries = {
...
+    "encounter": {
+      1: "The future you want, or the future I want... Let us see which one is more deserving, shall we?"
+    },
+    "victory": {
+      1: "Whaugh!"
+    },
+    "defeat": {
+      1: "Fools with no vision will continue to befoul this beautiful world."
+    }
+  },
```
Comment (josericardo-fo) [inline src/locales/pt_BR/dialogue.ts:565]: Suggested portuguese translations
```suggestion
  "rocket_grunt": {
    "encounter": {
      1: "Se prepara pra encrenca!"
    },
    "victory": {
      1: "Equipe Rocket decolando de novo!"
    },
  },
  "magma_grunt": {
    "encounter": {
      1: "Se você se meter com a Equipe Magma, não teremos piedade!"
    },
    "victory": {
      1: "Ahn? Eu perdi?"
    },
  },
  "aqua_grunt": {
    "encounter": {
      1: "Não pegamos leve com quem se mete com a Equipe Aqua, nem mesmo crianças!"
    },
    "victory": {
      1: "Tá de brincadeira!"
    },
  },
  "galactic_grunt": {
    "encounter": {
      1: "Não mexe com a Equipe Galáctica!"
    },
    "victory": {
      1: "Fui amassado..."
    },
  },
  "plasma_grunt": {
    "encounter": {
      1: "Não toleramos pessoas que pensam diferente de nós!"
    },
    "victory": {
      1: "Plasmaaaaaaaaa!"
    },
  },
  "flare_grunt": {
    "encounter": {
      1: "A moda é a coisa mais importante pra gente!"
    },
    "victory": {
      1: "O futuro não parece brilhante pra mim."
    },
  },
  "rocket_boss_giovanni_1": {
    "encounter": {
      1: "Tenho que admitir, estou impressionado que tenha chegado até aqui!"
    },
    "victory": {
      1: "QUÊ! Isso não é possível!"
    },
    "defeat": {
      1: "Guarde minhas palavras.\nNão ser capaz de medir sua própria força mostra que você ainda é uma criança."
    }
  },
  "rocket_boss_giovanni_2": {
    "encounter": {
      1: "Meus antigos associados precisam de mim... Você vai ficar no meu caminho?"
    },
    "victory": {
      1: "Como isso é possível...?\nO precioso sonho da Equipe Rocket se tornou pouco mais que uma ilusão..."
    },
    "defeat": {
      1: "A Equipe Rocket renascerá, e eu dominarei o mundo!"
    }
  },
  "magma_boss_maxie_1": {
    "encounter": {
      1: "Eu vou te enterrar com minhas próprias mãos.\nEspero que você aprecie essa honra!"
    },
    "victory": {
      1: "Ugh! Você é... bastante capaz...\nEu fiquei para trás, mas apenas por um triz..."
    },
    "defeat": {
      1: "A Equipe Magma vai prevalecer!"
    }
  },
  "magma_boss_maxie_2": {
    "encounter": {
      1: "Você é o último obstáculo entre mim e meus objetivos.\nPrepare-se para meu ataque final! Fuhahaha!"
    },
    "victory": {
      1: "Isso... Isso não é... Ngh..."
    },
    "defeat": {
      1: "E agora... Eu transformarei este planeta em uma terra ideal para a humanidade."
    }
  },
  "aqua_boss_archie_1": {
    "encounter": {
      1: "Eu sou o líder da Equipe Aqua, então temo que esse seja o fim da linha para você."
    },
    "victory": {
      1: "Vamos nos encontrar de novo em algum lugar. Eu vou ter certeza de lembrar desse rosto."
    },
    "defeat": {
      1: "Brilhante! Nada vai parar minha equipe agora!"
    }
  },
  "aqua_boss_archie_2": {
    "encounter": {
      1: "Estive esperando tanto tempo por este dia.\nEste é o verdadeiro poder da minha equipe!"
    },
    "victory": {
      1: "Como eu suspeitava..."
    },
    "defeat": {
      1: "Eu vou voltar tudo neste mundo ao seu estado puro e original!!"
    }
  },
  "galactic_boss_cyrus_1": {
    "encounter": {
      1: "Você foi compelido a vir aqui por tal sentimentalismo vazio\nEu farei você se arrepender de ter ouvido seu coração!"
    },
    "victory": {
      1: "Interessante. E bastante curioso."
    },
    "defeat": {
      1: "Eu criarei meu novo mundo..."
    }
  },
  "galactic_boss_cyrus_2": {
    "encounter": {
      1: "Nos encontramos novamente. Parece que nossos destinos estão entrelaçados.\nMas aqui e agora, eu finalmente quebrarei esse vínculo!"
    },
    "victory": {
      1: "Como? Como? COMO?!"
    },
    "defeat": {
      1: "Até logo."
    }
  },
  "plasma_boss_ghetsis_1": {
    "encounter": {
      1: "Ninguém pode me deter! Não importa quem seja ou o que faça!"
    },
    "victory": {
      1: "Como isso é possível? Eu sou o criador da Equipe Plasma! Eu sou perfeito!"
    },
    "defeat": {
      1: "Eu sou o governante perfeito de um novo mundo perfeito! Mwa ha ha!"
    }
  },
  "plasma_boss_ghetsis_2": {
    "encounter": {
      1: "Vamos! Eu quero ver sua cara depois que você perder toda a esperança!"
    },
    "victory": {
      1: "Meus cálculos... Não! Meus planos cuidadosos! O mundo deveria ser meu!"
    },
    "defeat": {
      1: "Kyurem! Use Absofusion!"
    }
  },
  "flare_boss_lysandre_1": {
    "encounter": {
      1: "Você está aqui para me deter? Mostre-me em batalha."
    },
    "victory": {
      1: "Você está aqui para me deter. Mas eu peço que você espere."
    },
    "defeat": {
      1: "Pokémon... não devem mais existir."
    }
  },
  "flare_boss_lysandre_2": {
    "encounter": {
      1: "O futuro que você quer, ou o futuro que eu quero... Vamos ver qual é o mais merecedor, não é mesmo?"
    },
    "victory": {
      1: "Uau!"
    },
    "defeat": {
      1; "Tolos sem visão continuarão a poluir este belo mundo."
    }
  },
```

**Review by brain-frog — APPROVED** [review]

**Pull request merged**